### ISSUE TITLE
Brief immunity period after damage

### ISSUE THREAD

**Issue by Forthoney** [Forthoney/recurring-nightmare#55]
**DISCUSS FURTHER WITH DESIGN TEAM**
As mentioned on discord, a problem that many testers had during the first playtest was that if an enemy fell on them or they fell on an enemy, there was no easy way to actually separate from the enemy, since the enemy's tracking is instantaneous.

So I think we should consider giving the player immunity for a bit after they get hit. During this period, the player would also not collide with the enemy. Think of the blinking effect in some games after you get hit.

We can also consider staggering an enemy that has hit a player, where for a split second after they make contact with a player, they will stop tracking the player (only for the enemy that hit the player though. Other enemies go on about their day)

**Comment by nightflowering** [comment]
Updated PlayerHealth script so that DamagePlayer() can make the player immune for a few seconds. The player is also immune at the start of the game, which we can change.
If we want to make the player not collide with enemies during the immunity period, we can change the player to another layer that is set to collide with the terrain but not enemies (Edit->Project Settings->Physics 2D->Layer Collision Matrix). I tried it, and it's a little weird, as the player will collide with the enemy for one frame and then phase through, which just looks like a little stutter.
Possible alternatives: Player never physically collides with enemies. Player bounces off somehow (could also maybe just bounce off the top but phase through the sides). Player can jump while on top of enemy.
I've found that making the enemy move slower also helps solve the problem of the player getting stuck on top of enemies.
